distribution/src/main/assembly/dist.xml

```diff
@@ -20,254 +20,139 @@
-                            <include>org.javassist:javassist</include>
-                            <include>org.jboss.logging:jboss-logging</include>
-                            <include>org.jboss.logging:jboss-logging-annotations</include>
-                            <include>org.apache.lucene:lucene-core</include>
-                            <include>org.apache.lucene:lucene-facet</include>
-                            <include>com.thoughtworks.paranamer:paranamer</include>
-                            <include>org.slf4j:slf4j-api</include>
-                            <include>org.antlr:stringtemplate</include>
-                            <include>org.jgroups:jgroups</include>
+        </dependencySet>
+
+        <!-- Optional libraries -->
+        <dependencySet>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>dist/lib/optional</outputDirectory>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.hibernate:hibernate-search-orm</include>
+                 <include>org.hibernate.hql:hibernate-hql-lucene</include>
+            </includes>
+        </dependencySet>
+
+        <dependencySet>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>dist/lib/optional/parboiled-java</outputDirectory>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.parboiled:parboiled-java</include>
+            </includes>
+        </dependencySet>
 
-                            <!-- Parboiled and deps -->
-                            <include>org.parboiled:parboiled-java</include>
-                            <include>org.parboiled:parboiled-core</include>
-                            <include>org.ow2.asm:asm</include>
-                            <include>org.ow2.asm:asm-tree</include>
-                            <include>org.ow2.asm:asm-analysis</include>
-                            <include>org.ow2.asm:asm-util</include>
-                        </includes>
-                    </dependencySet>
-                    <dependencySet>
-                        <useProjectArtifact>false</useProjectArtifact>
-                        <outputDirectory>/dist/lib/provided</outputDirectory>
-                        <scope>provided</scope>
-                    </dependencySet>
-                </dependencySets>
-            </binaries>
-        </moduleSet>
+        <!-- Provided libraries -->
+        <dependencySet>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>dist/lib/provided</outputDirectory>
+            <useTransitiveFiltering>false</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.jboss.logging:jboss-logging-annotations</include>
```

Comment:
We don't need the annotations. They are only needed at OGM's build time (we somewhat ab-use "provided" to emulate a "compile-only" scope for this).